token/typed.go

```diff
@@ -0,0 +1,59 @@
+package token
```

Comment:
The argument for the empty package is very true. The argument about the functions is also true, but godoc is already too crowded when it comes to the `token` package. Do not get me wrong, my API-mind also screams to put this into `token/typed` but the Go-way is a bit different. Have a look at the `io` package. It seems also crowded. So I think it is OK to put this into `token` since it resides in its own file there. And I rather have another token specific API in the token package than a pattern breakage.
There is however a problem with the `token` package. I put some things there that should be somewhere else, e.g., the parser stuff. I need to refactore the package but that is another issue and does not concern your PR.